### PR TITLE
Sending emails from irc is a good idea

### PR DESCRIPTION
**Pull request by LinuxMercedes**
I think SES handles rate limiting, so that should be OK.

### CODE REVIEW

**Comment by euank** [comment]
.. I think you might have gone too far.

**Comment by streed** [comment]
Squash then LGTM +1

**Comment by cbgbt** [comment]
Man, I hate being irc-primary.

**Comment by brhoades** [comment]
Sounds exciting to have on someone else's server. +1

File: conf/page.example.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	"secretAccessKey": "sekretz",
+	"region": "from http://docs.aws.amazon.com/ses/latest/DeveloperGuide/regions.html",
+	"host":"mx.cool.io",
+	"port":465,
+	"auth": {
+		"user": "totally@cool.io",
+		"pass": "hunter2"
+	},
+	"debug": false,
+	"secure": false,
```
Comment (hfukada) [inline conf/page.example.json]: i feel triggered

**Comment by hfukada** [comment]
this is a great idea +1

File: conf/page.example.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+	"accessKeyId": "keyz",
+	"secretAccessKey": "sekretz",
```
Comment (streed) [inline conf/page.example.json]: Inconsistent spelling of `secret`

Comment (LinuxMercedes) [inline conf/page.example.json]: nerd

File: conf/page.example.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	"region": "from http://docs.aws.amazon.com/ses/latest/DeveloperGuide/regions.html",
+	"host":"mx.cool.io",
+	"port":465,
+	"auth": {
+		"user": "totally@cool.io",
+		"pass": "hunter2"
+	},
+	"debug": false,
+	"secure": false,
+	"authMethod": "PLAIN",
```
Comment (streed) [inline conf/page.example.json]: I would have preferred you said "Conservative" here.

File: conf/page.example.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	"ignoreTLS": false,
+	"tls": {
+		"rejectUnauthorized": true
+	},
+	"from": "The Sun's Shades <totally@cool.io>",
+	"subject": "You're Outta Sight!"
```
Comment (streed) [inline conf/page.example.json]: :+1:

File: conf/page.example.json (hunk 1 above)
Comment (LinuxMercedes) [inline conf/page.example.json]: 🔫

File: conf/page.example.json (hunk 3 above)
Comment (LinuxMercedes) [inline conf/page.example.json]: *PLANE

File: conf/page.example.json (hunk 4 above)
Comment (LinuxMercedes) [inline conf/page.example.json]: 😎

File: modules/page/index.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+module.exports.commands = {
+  subscribe: function(r, p, reply, command, from, channel) {
+    if(p.length >= 1) {
+      getAddresses(channel, function() {
+        if(ses) {
+          ses.verifyEmailIdentity({'EmailAddress': p[0]}, function(err, data) {
+            if(err) {
+              console.log("SES email verify error " + err);
```
Comment (streed) [inline modules/page/index.js]: I suggest leaking as much information about the SES configuration to the user here.

**Comment by streed** [comment]
:shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit: :shipit:

File: modules/page/index.js (hunk 5 above)
Comment (LinuxMercedes) [inline modules/page/index.js]: For "debugging purposes only", of course.

**Comment by gmackie** [comment]
:8ball: cocaine.

**Comment by drusepth** [comment]
:santa:

**Comment by streed** [comment]
Customers are asking for this feature now, when can we expect this to ship?

File: modules/page/index.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        }
+      });
+    }
+    else {
+      reply("you didn't tell me enough stuff, doofus");
+    }
+  },
+  
+  unsubscribe: function(r, p, reply, command, from, channel) {
+    delete addresses[channel][from];
```
Comment (drusepth) [inline modules/page/index.js]: Did you mean to make this function actually do anything?

File: modules/page/index.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+          reply("Added address " + p[0] + " for user " + from);
+        }
+      });
+    }
+    else {
+      reply("you didn't tell me enough stuff, doofus");
+    }
+  },
+  
+  unsubscribe: function(r, p, reply, command, from, channel) {
```
Comment (drusepth) [inline modules/page/index.js]: This should be in all caps.

**Comment by gmackie** [comment]
Ship it

File: modules/page/index.js (hunk 6 above)
Comment (LinuxMercedes) [inline modules/page/index.js]: Yeah, it's supposed to delete you from the address list so people can't page you.

Comment (drusepth) [inline modules/page/index.js]: That seems odd for a function named UNSUBSCRIBE

**Pull request merged**